app/src/main/java/com/omkarmoghe/pokemap/controllers/MarkerRefreshController.java

```diff
@@ -64,44 +62,55 @@ public static MarkerRefreshController getInstance() {
 
 
     /**
-     * Cleanup thread state.
-     * Must ensure thread is properly quitting to allow safe reInit when needed
+     * Cleanup Messages and cancels the timer if it is running.
      */
     public void clear() {
-        if(mThread!=null) {
-            try {
-                mThread.join();
-                mThread.quit();            //I don't need to process the scheduled messages in queue.
-
-            } catch (InterruptedException e) {
-                Log.d(TAG, "Quitting too soon.. doing final cleanup");
-            } finally {
-                mThread = null;
-                mInstance = null;
-            }
+        if(mTimer != null){
+            mTimer.cancel();
+            mTimer = null;
         }
+        mHandler.removeMessages(MARKER_EXPIRED);
+
     }
 
-    /**
-     * Combo call for starting, resetting, refreshing, scheduling next update
-     * Must be called separately after [@link notifyTimeToExpiry]
-     */
-    public void reset() {
-        getInstance().mHandler.removeCallbacks(markerExpiryUpdate);
-        getInstance().mHandler.postDelayed(markerExpiryUpdate, updateInterval);
-        updateInterval = DEFAULT_UPDATE_INTERVAL;
+    public void startTimer(PokemonMarkerExtended markerData){
+        if(mTimer != null){
+            mTimer.cancel();
+        }
+
+        long time = markerData.getCatchablePokemon().getExpirationTimestampMs() - System.currentTimeMillis();
+        final MarkerUpdate event = new MarkerUpdate(markerData);
+        mTimer = new CountDownTimer(time, DEFAULT_UPDATE_INTERVAL) {
+            @Override
+            public void onTick(long l) {
+                EventBus.getDefault().post(event);
```

Comment:
If timer ticks are needed for updating the snippet on marker, only run the timer tick for markers with visible infoWindows.